crates/core/src/delta_datafusion/table_provider.rs

```diff
@@ -442,13 +450,30 @@ impl<'a> DeltaScanBuilder<'a> {
 
         let stats = stats.unwrap_or(Statistics::new_unknown(&schema));
 
-        let parquet_options = TableParquetOptions {
-            global: self.session.config().options().execution.parquet.clone(),
-            ..Default::default()
-        };
+        let parquet_options = self
+            .parquet_options
+            .unwrap_or_else(|| self.session.table_options().parquet.clone());
+
+        // We have to set the encryption factory on the ParquetSource based on the Parquet options,
+        // as this is usually handled by the ParquetFormat type in DataFusion,
+        // which is not used in delta-rs.
```

Comment:
If we pass a `FileFormatOptions` instead of `TableParquetOptions`, we could maybe add a new method to `FileFormatOptions` to allow customizing the `ParquetSource` creation using and avoid needing to set the encryption factory here.

Although I don't feel like this code is too bad but it does leak some knowledge of Parquet encryption outside of the `FileFormatOptions` implementation.